projects/plugins/jetpack/modules/scan/class-admin-sidebar-link.php

```diff
@@ -65,21 +65,25 @@ public function maybe_add_admin_link() {
 			return;
 		}
 
-		$has_scan   = $this->has_scan();
-		$has_backup = $this->has_backup();
+		$has_scan          = $this->has_scan();
+		$has_backup        = $this->has_backup();
+		$has_backup_plugin = $this->has_backup_plugin();
+		$menu_label        = '';
 
 		$url = Redirect::get_url( 'calypso-backups' );
-		if ( $has_scan && ! $has_backup ) {
+		if ( $has_scan && ! $has_backup || $has_scan && $has_backup && $has_backup_plugin ) {
 			$menu_label = __( 'Scan', 'jetpack' );
 			$url        = Redirect::get_url( 'calypso-scanner' );
-		} elseif ( ! $has_scan && $has_backup ) {
+		} elseif ( ! $has_scan && $has_backup && ! $has_backup_plugin ) {
 			$menu_label = __( 'Backup', 'jetpack' );
-		} else {
-			// Will be both, as the code won't get this far if neither is true.
+		} elseif ( $has_scan && $has_backup && ! $has_backup_plugin ) {
 			$menu_label = __( 'Backup & Scan', 'jetpack' );
 		}
 
```

Comment:
This section is getting more and more complicated. What would you think about adding a comment above it, explaining the different possible scenarios we're trying to cover?

Reply:
Agree. Instead of adding comments, I made the code more readable. See what you think

Reply:
Yeah, that's indeed a lot better, thank you!